bottlecap/src/proc/mod.rs

```diff
@@ -0,0 +1,82 @@
+pub mod constants;
+
+use std::{
+    error::Error,
+    fs::File,
+    io::{self, BufRead},
+};
+
+use constants::{LAMDBA_NETWORK_INTERFACE, PROC_NET_DEV_PATH};
+
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub struct NetworkData {
+    pub rx_bytes: f64,
+    pub tx_bytes: f64,
+}
+
+pub fn get_network_data() -> Result<NetworkData, Box<dyn std::error::Error>> {
+    get_network_data_helper(PROC_NET_DEV_PATH)
+}
+
+fn get_network_data_helper(path: &str) -> Result<NetworkData, Box<dyn std::error::Error>> {
+    let file = File::open(path).map_err(|e| Box::new(e) as Box<dyn Error>)?;
+    let reader = io::BufReader::new(file);
+
+    for line in reader.lines() {
+        let line = line.map_err(|e| Box::new(e) as Box<dyn Error>)?;
+        let mut values = line.split_whitespace();
+
+        let Some(interface_name) = values.next() else {
+            continue;
+        };
+        if !interface_name.starts_with(LAMDBA_NETWORK_INTERFACE) {
+            continue;
+        }
+
+        let rx_bytes: f64 = match values.next().and_then(|s| s.parse().ok()) {
+            Some(value) => value,
+            None => continue,
+        };
+        let tx_bytes: f64 = match values.nth(7).and_then(|s| s.parse().ok()) {
+            Some(value) => value,
+            None => continue,
+        };
+
+        return Ok(NetworkData { rx_bytes, tx_bytes });
```

Comment:
I'm not sure I understand the logic, as I'm not familiar with it, could you make it more readable, or add some code comments in order to help the reader understand why you are doing this? Or document what this method does? 

Someone with no context would struggle with this code